File: decision-aide-app/backend/src/main/java/sn/siad/model/Utilisateur.java
package sn.siad.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.Instant;

@Entity
@Table(name = "utilisateurs")
@Getter
@Setter
public class Utilisateur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nomComplet;
    @Column(unique = true)
    private String email; // pour enseignants, admin, inspection, ministere

    @Column(unique = true)
    private String matricule; // pour eleves

    private String motDePasse;

    @Enumerated(EnumType.STRING)
    private RoleUtilisateur role;

    private boolean actif = true;

    @ManyToOne
    private Etablissement etablissement;

    private Instant creeLe = Instant.now();
}

File: decision-aide-app/frontend/lib/modeles/utilisateur.dart
class UtilisateurModele {
  final int id;
  final String nomComplet;
  final String? email;
  final String? matricule;
  final String role;
  final bool actif;
  final String? etablissementNom;

  UtilisateurModele({
    required this.id,
    required this.nomComplet,
    this.email,
    this.matricule,
    required this.role,
    required this.actif,
    this.etablissementNom,
  });

  factory UtilisateurModele.fromJson(Map<String, dynamic> json) => UtilisateurModele(
        id: json['id'] as int,
        nomComplet: json['nomComplet'] as String? ?? '',
        email: json['email'] as String?,
        matricule: json['matricule'] as String?,
        role: json['role'] as String? ?? '',
        actif: json['actif'] as bool? ?? false,
        etablissementNom: json['etablissement']?['nom'] as String?,
      );
}
